### PR TITLE
[SQL] support MATERIALIZED table and views

### PR DESCRIPTION
**Pull request by mihaibudiu**
Remaining tasks:
- [x] Materialization strategy for tables
- [x] WebConsole update: grey out the VIEW button for non-materialized tables and views (@Karakatiza666)
- [x] Update demos, making a subset of tables/views MATERIALIZED
- [x] Update docs

Is this a user-visible change (yes/no): yes

Addresses #1920

### CODE REVIEW

File: sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/RegresssionTests.java [hunk 1]
```diff
@@ -138,6 +141,30 @@ public void missingCast() {
         this.compileRustTestCase(sql);
     }
 
+    @Test @Ignore("Calcite decorrelator fails")
```
Comment (ryzhyk) [inline sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/RegresssionTests.java:144]: This doesn't seem to belong here.

Comment (mihaibudiu) [reply sql-to-dbsp-compiler/SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/sql/simple/RegresssionTests.java:144]: It has to be checked in at some point

**Review by mihaibudiu — COMMENTED** [review]
looks fine

**Comment by ryzhyk** [comment]
@Karakatiza666 , required WebConsole changes:
- To determine whether a table or view is materialized, use the `materialized` attribute in the JSON schema.
- For non-materialized tables and views, grey-out the eye icons and the BROWSE tab.
  - When hovering over the greyed-out view, display the following hint: "Use 'CREATE MATERIALIZED VIEW' syntax to enable browsing the view".
  - When hovering over the greyed-out table, display the following hint: "Use 'CREATE TABLE (..) WITH (materialized=true)' syntax to enable browsing the table".

**Comment by Karakatiza666** [comment]
image
image

**Pull request merged**